src/FullTextSearchHelpers.ts

```diff
@@ -0,0 +1,118 @@
+    'ě': 'e', 'Ĝ': 'G', 'ĝ': 'g', 'Ğ': 'G', 'ğ': 'g', 'Ġ': 'G', 'ġ': 'g', 'Ģ': 'G', 'ģ': 'g', 'Ĥ': 'H',
+    'ĥ': 'h', 'Ĩ': 'I', 'ĩ': 'i', 'Ī': 'I', 'ī': 'i', 'Ĭ': 'I', 'ĭ': 'i', 'Į': 'I', 'į': 'i', 'İ': 'I',
+    'Ĳ': 'IJ', 'ĳ': 'ij', 'Ĵ': 'J', 'ĵ': 'j', 'Ķ': 'K', 'ķ': 'k', 'Ĺ': 'L', 'ĺ': 'l', 'Ļ': 'L', 'ļ': 'l',
+    'Ľ': 'L', 'ľ': 'l', 'Ŀ': 'L·', 'ŀ': 'l·', 'Ń': 'N', 'ń': 'n', 'Ņ': 'N', 'ņ': 'n', 'Ň': 'N', 'ň': 'n',
+    'ŉ': 'ʼn', 'Ō': 'O', 'ō': 'o', 'Ŏ': 'O', 'ŏ': 'o', 'Ő': 'O', 'ő': 'o', 'Ŕ': 'R', 'ŕ': 'r', 'Ŗ': 'R',
+    'ŗ': 'r', 'Ř': 'R', 'ř': 'r', 'Ś': 'S', 'ś': 's', 'Ŝ': 'S', 'ŝ': 's', 'Ş': 'S', 'ş': 's', 'Š': 'S',
+    'š': 's', 'Ţ': 'T', 'ţ': 't', 'Ť': 'T', 'ť': 't', 'Ũ': 'U', 'ũ': 'u', 'Ū': 'U', 'ū': 'u', 'Ŭ': 'U',
+    'ŭ': 'u', 'Ů': 'U', 'ů': 'u', 'Ű': 'U', 'ű': 'u', 'Ų': 'U', 'ų': 'u', 'Ŵ': 'W', 'ŵ': 'w', 'Ŷ': 'Y',
+    'ŷ': 'y', 'Ÿ': 'Y', 'Ź': 'Z', 'ź': 'z', 'Ż': 'Z', 'ż': 'z', 'Ž': 'Z', 'ž': 'z', 'ſ': 's', 'Ơ': 'O',
+    'ơ': 'o', 'Ư': 'U', 'ư': 'u', 'Ǆ': 'DZ', 'ǅ': 'Dz', 'ǆ': 'dz', 'Ǉ': 'LJ', 'ǈ': 'Lj', 'ǉ': 'lj',
+    'Ǌ': 'NJ', 'ǋ': 'Nj', 'ǌ': 'nj', 'Ǎ': 'A', 'ǎ': 'a', 'Ǐ': 'I', 'ǐ': 'i', 'Ǒ': 'O', 'ǒ': 'o', 'Ǔ': 'U',
+    'ǔ': 'u', 'Ǖ': 'U', 'ǖ': 'u', 'Ǘ': 'U', 'ǘ': 'u', 'Ǚ': 'U', 'ǚ': 'u', 'Ǜ': 'U', 'ǜ': 'u', 'Ǟ': 'A',
+    'ǟ': 'a', 'Ǡ': 'A', 'ǡ': 'a', 'Ǧ': 'G', 'ǧ': 'g', 'Ǩ': 'K', 'ǩ': 'k', 'Ǫ': 'O', 'ǫ': 'o', 'Ǭ': 'O',
+    'ǭ': 'o', 'ǰ': 'j', 'Ǳ': 'DZ', 'ǲ': 'Dz', 'ǳ': 'dz', 'Ǵ': 'G', 'ǵ': 'g', 'Ǹ': 'N', 'ǹ': 'n', 'Ǻ': 'A',
+    'ǻ': 'a', 'Ȁ': 'A', 'ȁ': 'a', 'Ȃ': 'A', 'ȃ': 'a', 'Ȅ': 'E', 'ȅ': 'e', 'Ȇ': 'E', 'ȇ': 'e', 'Ȉ': 'I',
+    'ȉ': 'i', 'Ȋ': 'I', 'ȋ': 'i', 'Ȍ': 'O', 'ȍ': 'o', 'Ȏ': 'O', 'ȏ': 'o', 'Ȑ': 'R', 'ȑ': 'r', 'Ȓ': 'R',
+    'ȓ': 'r', 'Ȕ': 'U', 'ȕ': 'u', 'Ȗ': 'U', 'ȗ': 'u', 'Ș': 'S', 'ș': 's', 'Ț': 'T', 'ț': 't', 'Ȟ': 'H',
+    'ȟ': 'h', 'Ȧ': 'A', 'ȧ': 'a', 'Ȩ': 'E', 'ȩ': 'e', 'Ȫ': 'O', 'ȫ': 'o', 'Ȭ': 'O', 'ȭ': 'o', 'Ȯ': 'O',
+    'ȯ': 'o', 'Ȱ': 'O', 'ȱ': 'o', 'Ȳ': 'Y', 'ȳ': 'y', 'ʰ': 'h', 'ʲ': 'j', 'ʳ': 'r', 'ʷ': 'w', 'ʸ': 'y',
+    'ˡ': 'l', 'ˢ': 's', 'ˣ': 'x'
+};
+const _minLead = parseInt('0xD800', 16);
+const _maxLead = parseInt('0xDBFF', 16);
+
+export function normalizeString(input: string): string {
+    // Constants for leading byte-range.
+    let currentChar = '';
+    let output = '';
+
+    if (!input) {
+        return input;
+    }
+
+    for (var i = 0; i < input.length; i++) {
+        currentChar += input.charAt(i);
+        const code = input.charCodeAt(i);
+        if (code < _minLead || code > _maxLead) {
+            output += _charMap[currentChar] ? _charMap[currentChar] : currentChar;
+            currentChar = '';
+        }
+    }
+
+    return output;
+}
+
+export function breakWords(rawString: string): string[] {
+    // Figure out how to do this in a localized fashion
+    return rawString.split(' ');
+}
+
+export function breakAndNormalizeSearchPhrase(phrase: string): string[] {
+    return _.uniq(_.map(breakWords(phrase), word => normalizeSearchTerm(word)));
```

Comment:
I think you want `_.uniqBy` since `_.uniq` only looks at its first argument (as of lodash 4).